File: pathmind-webapp/src/main/java/io/skymind/pathmind/webapp/ui/components/molecules/CookieConsentBox.java
package io.skymind.pathmind.webapp.ui.components.molecules;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.CssImport;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;

@Tag("cookie-consent-box")
@CssImport("./styles/components/cookie-consent.css")
@JsModule("./src/components/molecules/cookie-consent-box.ts")
public class CookieConsentBox extends LitTemplate {
}

File: pathmind-webapp/frontend/src/components/molecules/cookie-consent-box.ts
import { LitElement, html, css } from "lit-element";
import "cookieconsent";

class CookieConsentBox extends LitElement {
    static get styles() {
        return css`
            :host {
                display: none;
            }
        `
    }

    render() {
        return html``;
    }

    constructor() {
        super();
        if (document.querySelectorAll(".cc-window[aria-label~='cookieconsent']").length === 0) {
            (window as any).cookieconsent.initialise({
                container: document.querySelector("vaadin-app-layout"),
                content: {
                  message: "This website uses cookies to ensure you get the best experience.",
                  dismiss: "Got it!",
                  link: "Learn more",
                  href: "https://pathmind.com/privacy"
                },
                position: "bottom-left",
            });
        }
    }
}

customElements.define("cookie-consent-box", CookieConsentBox);